### ISSUE TITLE
gpu-switch -i causes computer to just freeze on osx 10.12.4

### ISSUE THREAD

**Issue by adityaravishankar** [0xbb/gpu-switch#39]
Just got the latest code off github and ran the build script.

Freshly booted laptop with gfxCardStatus running in the background.

Running ./gpu-switch -i caused the computer to freeze completely. 

Any advice would be helpful. 

Model: MacBookPro8,2
OS: macOS Sierra 10.12.4

Happy to share/add further debugging/logging information  as needed, to help with fixing this issue.

**Comment by 0xbb** [comment]
I don't have access to a MacBook anymore and I can't help you with this issue. 
But I am not sure if gpu-switch should be run together with gfxCardStatus.

**Comment by rythmkraze** [comment]
Same issue here. Running `./gpu-switch -i` freezes the computer. No gfxCardStatus running in the background. But gfxCardStatus also exhibits the same behaviour, that is switching to "integrated only" freezes the computer.

Model Identifier:	MacBookPro8,2
OS: macOS Sierra  10.12.4 (16E195)

**Comment by ah-** [comment]
I suspect this is due to changes Apple made in 10.12.4. switcher/gfxCardStatus call a private API, and the one to switch GPU in particular hasn't been in use by the official energy saver preference pane for a long time now (if it ever was, don't quite remember).

I guess this call is just broken now and there's not a whole lot we can do about it. Someone with a lot of time would need to go and look if there's a new alternative way to switch to internal only mode.

**Comment by rythmkraze** [comment]
Thanks @ah- for the quick response. Looks like I am out of luck then :(

**Comment by codedmart** [comment]
Not sure if my issue is related. I have been using gpu-switch for a while just fine, but noticed since the upgrading to 10.12.4 I can't get my integrated card to work in Linux. I see it as powered off when I `cat` `vgaswitcheroo`, but if I try to switch to it I just get a blank screen now. When I run `./gpu-switch -i` my system doesn't freeze though.

**Comment by codedmart** [comment]
@0xbb Anyway I can help with getting this to work with Sierra 10.12.4?

**Comment by codedmart** [comment]
@ah- Any way I can help with progress on this?

**Comment by ah-** [comment]
@codedmart could you try getting some log output from dmesg during/after the switch to a blank screen? Either by ssh-ing in or switching back blindly?
Ideally with drm.debug=0x06 set to get a bit more verbose output if there's nothing clear in dmesg otherwise.

And so gpu-switch -i still works for you (on linux)? Does it actually boot up with the integrated card?

**Comment by codedmart** [comment]
@ah- I don't get any errors in OSX when running `gpu-switch -i`, and when I boot into linux I do see the integrated card, but I can't get it to work anymore (just the blank screen). It use to work fine before 10.12.4. I will try to dump some output from dmesg.

**Comment by ah-** [comment]
Has your linux kernel changed since it worked?

I'm just looking for any hint in the logs if it detects a screen at all and if the resolution is correct etc.

**Comment by codedmart** [comment]
I don't think it is kernel related because I tried various kernels I had between upgrades, but I could be wrong. I will get more output as soon as I can.
